1_preprocessing_data/PyBaSiC_Pipelines/scripts/nbconverted/Illumination_Correction.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Illumination Correction of NF1 Images
+
+# ## Import libraries
+
+# In[1]:
+
+
+import sys
+import numpy as np
+import matplotlib
+import pathlib
+from matplotlib import pyplot as plt
+from pathlib import Path
+import os
+import skimage
+
+# explicit import to PyBaSiC due to not having package support
+sys.path.append("./PyBaSiC/")
+import pybasic
+
+
+# ## Create path and load in images
+# The directory set up used for this experiment goes as such:
+# ```
+# 
+# ├── NF1_SchwannCell_data/
+# │   ├── 1_preprocessing_data/
+# │   │   ├── PyBaSiC_Pipelines/
+# │   │   │   ├── NF1_Channels/
+# │   │   │   │   ├── DAPI/ or GFP/ or RFP/
+# 
+# ```
+
+# In[2]:
+
+
+channel_path = '../PyBaSiC_Pipelines/NF1_Channels/RFP/'
+
+# The for loop is running through the files within the folder designated by channel_path and will only take the names of the images and strips the ".tif"
+# which leaves the image name that helps with identification of cells (genotype)
+
+all_files = os.listdir(channel_path)
+file_ext = ".tif"
+image_files = []
+for file in all_files:
+    if file.endswith(file_ext):
+        image_files.append(file.strip(file_ext))
+image_files.sort()
+channel_images = pybasic.tools.load_data(channel_path, '.tif', verbosity = True)
```

Comment:
Any time we notice this pattern that we'll use more than once, we should think "yes! we can turn this into a function"

Something like:

```python
def load_pybasic_data(channel_path, channel, file_extension=".tif"):
    """
    Load all images from a given directory in preparation for pybasic illum correction

    Parameters
    -----------
    channel_path: Pathlib.Path
        The directory where the channel folders are stored
    channel: str
        The name of the channel (currently one of ["GFP", "RFP", "DAPI"]
    file_extension: str
        The filename extension of the types of files to search for (default: ".tif")
    """

    full_channel_path = pathlib.Path(channel_path, channel)

    image_files = []
    for image_file in full_channel_path.iterdir():
        image_file = str(image_file)
        if image_file.endswith(file_extension):
            image_files.append(file.strip(file_extension))
   
    image_files.sort()
    channel_images = pybasic.tools.load_data(full_channel_path, file_extension, verbosity = True)

    return channel_images
```

A couple more notes:

- The current pipeline needs to be rerun multiple times, depending on the input channel. This is typically ok, but it is not fully reproducible. However, we can achieve full reproducibilty in this notebook!
- When we write custom functions, we define them immediately after imports (PEP convention). So please move this to executed cell 2
- Note the docstring of the function. Spending a bit of time writing documentation for your functions will save you loads of time later trying to figure out what your functions are doing :) (I recommend doing so!)